public/framework/catalog_cache/cache_manager.ts

```diff
@@ -68,27 +74,34 @@ export class CatalogCacheManager {
    * Retrieves accelerations cache from local storage.
    * @returns {AccelerationsCacheData} The retrieved accelerations cache.
    */
-  static getAccelerationsCache(): AccelerationsCacheData {
+  static getAccelerationsCache(dataSourceMDSId?: string): AccelerationsCacheData {
     const accelerationCacheData = sessionStorage.getItem(this.accelerationsCacheKey);
 
     if (accelerationCacheData) {
-      return JSON.parse(accelerationCacheData);
-    } else {
-      const defaultCacheObject = {
-        version: CATALOG_CACHE_VERSION,
-        dataSources: [],
-      };
-      this.saveAccelerationsCache(defaultCacheObject);
-      return defaultCacheObject;
+      const cachedAcclerationData = JSON.parse(accelerationCacheData);
+      if (dataSourceMDSId && cachedAcclerationData.dataSourceMDSId === dataSourceMDSId) {
```

Comment:
do we need to trim 'dataSourceMDSId' ? what if value passed in is ' MDSId-123 ' and the actual id is 'MDSId-123'?

Reply:
I dont think we are adding a space to the id anywhere, could you clarify from where the space is?